File: tenspiler/llm/benchmarks/llama/source/rmsnorm/rmsnorm_part2.cc
#include <vector>
using namespace std;

int integer_sqrt(int x) { return x; }

vector<int> rmsnorm_part2(vector<int> input, vector<int> weight, int ss) {
    vector<int> output;
    int size = input.size();
    int inv_ss = 1 / integer_sqrt(ss / size + 1);
    for (int i = 0; i < input.size(); i++)
        output.push_back(inv_ss * input[i] * weight[i]);
    return output;
}

File: tenspiler/llama/python/for_synthesis/rmsnorm/rmsnorm_part2.py
from typing import List


def integer_sqrt(x: int) -> int:
    return x


def rmsnorm_part2_baseline(input: List[int], weight: List[int], ss: int) -> List[int]:
    output = []
    size = len(input)
    inv_ss = 1 / integer_sqrt(ss // size + 1)
    for i in range(size):
        output.append(inv_ss * input[i] * weight[i])
    return output
